cmake/TPLs/FindTPLSuperLU.cmake

```diff
@@ -56,5 +56,45 @@
 
 TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( SuperLU
   REQUIRED_HEADERS supermatrix.h slu_ddefs.h
-  REQUIRED_LIBS_NAMES "superlu superlu_3.0 superlu_4.0 superlu_4.1 superlu_4.2 superlu_4.3"
+  REQUIRED_LIBS_NAMES "superlu superlu_3.0 superlu_4.0 superlu_4.1 superlu_4.2 superlu_4.3 superlu_5.0"
```

Comment:
@bmpersc @jwillenbring @trilinos/framework Could you please review the changes in FindTPLSuperLU.cmake as provided by @bavier and merged here via PR #411 (with minor change to variable name)?
The purpose of this addition is to detect if the SuperLU version  >= 5.0. API changes in some SuperLU routines necessitate knowing this information; if version is >= 5.0 (as determined by the source code below properly compiling and running), then the variable `${PROJECT_NAME}_ENABLE_SuperLU5_API` is defined and used within if-guards in the affected codes to enable the proper SuperLU API after preprocessing.